ignite/metrics/metric.py

```diff
@@ -91,14 +98,14 @@ def compute(self) -> Any:
         """
         pass
 
-    def _sync_all_reduce(self, tensor: Union[torch.Tensor, numbers.Number]) -> Union[torch.Tensor, numbers.Number]:
-        if not (dist.is_available() and dist.is_initialized()):
-            # Nothing to reduce
-            return tensor
-
+    def _do_reduction(
+        self,
+        tensor: Union[torch.Tensor, numbers.Number],
+        reduction: Callable[[Union[torch.Tensor, numbers.Number]], None],
+    ) -> Union[torch.Tensor, numbers.Number]:
         tensor_to_number = False
         if isinstance(tensor, numbers.Number):
-            tensor = torch.tensor(tensor, device=self._device)
+            tensor = torch.tensor(tensor, device=self._device, dtype=torch.float)
```

Comment:
Passing to `dtype=torch.float` is only necessary for TPU as they do not support `torch.long` and probably others.
Maybe we can wait until XLA fix it and remove casting

Reply:
In the interim I've added an optional dtype arg to the `_do_reduction` method which is only populated in the TPU case.